Tasks/Kubernetes/Tests/L0.ts

```diff
@@ -0,0 +1,390 @@
+        delete process.env[shared.TestEnvVars.namespace];
+        delete process.env[shared.TestEnvVars.arguments];
+        delete process.env[shared.TestEnvVars.useConfigurationFile];
+        delete process.env[shared.TestEnvVars.secretName];
+        delete process.env[shared.TestEnvVars.forceUpdate];
+        delete process.env[shared.TestEnvVars.outputFormat];
+        delete process.env[shared.TestEnvVars.kubectlOutput];
+    });
+    after(function () {
+    });
+
+    it('Run sucessfully when the user provides a specific location for kubectl', (done:MochaDone) => {
+        let tp = path.join(__dirname, 'TestSetup.js');
+        let tr : ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+        process.env[shared.TestEnvVars.command] = shared.Commands.get;
+        process.env[shared.TestEnvVars.arguments] = "pods";    
+        process.env[shared.TestEnvVars.versionOrLocation] = "location";
+        process.env[shared.TestEnvVars.specifyLocation] = shared.formatPath("newUserDir/kubectl.exe");
+        process.env[shared.isKubectlPresentOnMachine] = "false";
+
+        tr.run();
+        assert(tr.succeeded, 'task should have succeeded');
+        assert(tr.invokedToolCount == 1, 'should have invoked tool one times. actual: ' + tr.invokedToolCount);
+        assert(tr.stderr.length == 0 || tr.errorIssues.length, 'should not have written to stderr');
+        assert(tr.stdout.indexOf(`[command]${shared.formatPath("newUserDir/kubectl.exe")} --kubeconfig ${shared.formatPath("newUserDir/config")} get pods`) != -1, "kubectl get should run");
+        console.log(tr.stderr);
+        done();
+    });
+
+    it('Run sucessfully when the user provides the version for kubectl with checkLatest as false', (done:MochaDone) => {
+        let tp = path.join(__dirname, 'TestSetup.js');
+        let tr : ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+        process.env[shared.TestEnvVars.command] = shared.Commands.get;
+        process.env[shared.TestEnvVars.arguments] = "pods";    
+        process.env[shared.TestEnvVars.versionOrLocation] = "version";
+        process.env[shared.TestEnvVars.versionSpec] = "1.7.0";
+        process.env[shared.isKubectlPresentOnMachine] = "false";
+        tr.run();
+
+        assert(tr.invokedToolCount == 1, 'should have invoked tool one times. actual: ' + tr.invokedToolCount);
+        assert(tr.stderr.length == 0 || tr.errorIssues.length, 'should not have written to stderr');
+        assert(tr.succeeded, 'task should have succeeded');
+        assert(tr.stdout.indexOf(`[command]${shared.formatPath("newUserDir/kubectl.exe")} --kubeconfig ${shared.formatPath("newUserDir/config")} get pods`) != -1, "kubectl apply should run");
+        console.log(tr.stderr);
+        done();
+    });
+
+    it('Run sucessfully when the user provides the version for kubectl with checkLatest as true', (done:MochaDone) => {
+        let tp = path.join(__dirname, 'TestSetup.js');
```

Comment:
I can not see any check that latest version is looked for and downloded.

Reply:
Added check